### PR TITLE
IBP-4209: moved 'MG_MANAGE_INVENTORY' permission to 'MANAGE_GERMPLASM'

### PR DESCRIPTION
**Pull request by lik-83**
Please, review

Include:
IntegratedBreedingPlatform/Workbench#434
IntegratedBreedingPlatform/Fieldbook#1018
IntegratedBreedingPlatform/BMSAPI#548

### CODE REVIEW

**Comment by clarysabel** [comment]
Hi @lik-83 
The changeset seems to be fine if we want the option to still be present in Germplasm List.
Though, We need to modify the Preauthorize in BMSAPI in PermissionsEnum.HAS_CREATE_LOTS_BATCH to also consider the new tree for the permission

**Review by clarysabel — CHANGES_REQUESTED** [review]
Hi @cuenyad 
I left some comments in this PR

File: src/main/java/org/generationcp/middleware/domain/inventory/common/SearchTypeComposeDto.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
+package org.generationcp.middleware.domain.inventory.common;
+
+
+public class SearchTypeComposeDto {
```
[the remark this answers is not in the document]
Comment (cuenyad) [reply src/main/java/org/generationcp/middleware/domain/inventory/common/SearchTypeComposeDto.java]: done.

File: src/main/resources/liquibase/workbench_changelog/18_3_0.xml [hunk 2]
```diff
@@ -24,4 +24,37 @@
 			INSERT INTO role_type_permission (role_type_id, permission_id, selectable) VALUES ('3', @view_change_history, '1');
 		</sql>
 	</changeSet>
+
+	<changeSet author="cuenyad" id="v18.3.0-2">
+        <preConditions onFail="MARK_RAN">
+            <sqlCheck expectedResult="0">
+                SELECT COUNT(*) FROM permission WHERE
+					(name = 'MG_MANAGE_INVENTORY' AND parent_id =
```
[the remark this answers is not in the document]
Comment (cuenyad) [reply src/main/resources/liquibase/workbench_changelog/18_3_0.xml]: done.

File: src/main/resources/liquibase/workbench_changelog/18_3_0.xml [hunk 3]
```diff
@@ -24,4 +24,37 @@
...
+					(SELECT permission_id FROM permission WHERE name = 'MANAGE_GERMPLASM'));
+            </sqlCheck>
+        </preConditions>
+        <comment>Update MG_MANAGE_INVENTORY permission</comment>
```
[the remark this answers is not in the document]
Comment (cuenyad) [reply src/main/resources/liquibase/workbench_changelog/18_3_0.xml]: done.

File: src/main/java/org/generationcp/middleware/domain/inventory/common/SearchTypeComposeDto.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+	private Integer searchRequestId;
+	private String searchType;
```
[the remark this answers is not in the document]
Comment (cuenyad) [reply src/main/java/org/generationcp/middleware/domain/inventory/common/SearchTypeComposeDto.java]: done.

File: src/main/java/org/generationcp/middleware/domain/inventory/common/SearchTypeComposeDto.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+	public String getSearchType() {
+		return searchType;
+	}
+
+	public void setSearchType(String searchType) {
+		this.searchType = searchType;
+	}
+
+	public enum SearchType {
+		MANAGE_STUDY("ManageStudy"), GERMPLASM_SEARCH("GermplasmSearch");
```
[the remark this answers is not in the document]
Comment (cuenyad) [reply src/main/java/org/generationcp/middleware/domain/inventory/common/SearchTypeComposeDto.java]: done.

**Pull request merged**